lib/instrumentation/modules/mongodb.js

```diff
@@ -4,185 +4,6 @@
-            } else {
-              if (!client[kListenersAdded]) {
-                client.on('commandStarted', onStart);
-                client.on('commandSucceeded', onEnd);
-                client.on('commandFailed', onEnd);
-                client[kListenersAdded] = true;
-              }
-              callback(err, client);
-            }
-          },
-        );
-      } else {
-        const p = origConnect.call(this, url, options, callback);
-        p.then((client) => {
-          if (!client[kListenersAdded]) {
-            client.on('commandStarted', onStart);
-            client.on('commandSucceeded', onEnd);
-            client.on('commandFailed', onEnd);
-            client[kListenersAdded] = true;
-          }
-        });
-        return p;
-      }
-    };
-  }
-
-  function onStart(event) {
-    // `event` is a `CommandStartedEvent`
-    // https://github.com/mongodb/specifications/blob/master/source/command-monitoring/command-monitoring.rst#api
-    // E.g. with mongodb@3.6.3:
-    //   CommandStartedEvent {
-    //     address: '127.0.0.1:27017',
-    //     connectionId: 1,
-    //     requestId: 1,
-    //     databaseName: 'test',
-    //     commandName: 'insert',
-    //     command:
-    //     { ... } }
-
-    const name = [
-      event.databaseName,
-      collectionFor(event),
-      event.commandName,
-    ].join('.');
-
-    const span = ins.createSpan(name, 'db', 'mongodb', event.commandName, {
-      exitSpan: true,
-    });
-    if (span) {
-      activeSpans.set(event.requestId, span);
-
-      // Destination context.
-      // Per the following code it looks like "<hostname>:<port>" should be
-      // available via the `address` or `connectionId` field.
-      // https://github.com/mongodb/node-mongodb-native/blob/dd356f0ede/lib/core/connection/apm.js#L155-L169
-      const address = event.address || event.connectionId;
-      let match;
-      if (
-        address &&
-        typeof address === 'string' &&
-        (match = HOSTNAME_PORT_RE.exec(address))
-      ) {
-        span._setDestinationContext(getDBDestination(match[1], match[2]));
-      } else {
-        agent.logger.trace(
-          'could not set destination context on mongodb span from address=%j',
-          address,
-        );
-      }
-
-      const dbContext = { type: 'mongodb', instance: event.databaseName };
-      span.setDbContext(dbContext);
-    }
-  }
-
-  function onEnd(event) {
-    if (!activeSpans.has(event.requestId)) return;
-    const span = activeSpans.get(event.requestId);
-    activeSpans.delete(event.requestId);
-    span.end(span._timer.start / 1000 + event.duration);
-  }
-
-  function collectionFor(event) {
-    const collection = event.command[event.commandName];
-    return typeof collection === 'string' ? collection : '$cmd';
-  }
-};
+// Since we are also instrumenting submodules we moved this instrumentation
+// to a dedicated folder
+module.exports = require('./mongodb/lib/index');
```

Comment:
Note to reviewer: moved to `modules/mongodb/lib/index.js` to keep all instrumentation logic together. The path is mimicking the one in the lib

Reply:
nit: I understand wanting to group all "mongodb-related instrumentation code" into one area. However, this currently makes the mongodb instrumentation special in that the patching of `require('mongodb')` is *not* at "lib/instrumentation/modules/mongodb.js".  Mostly, though, this (and the ESM changes, and the test file refactoring) is contributing to this being a rather large patch where the main point is the `AsyncResource.bind` usage.

Reply:
put back in place :)